src/ray/protobuf/experimental/instance_manager.proto

```diff
@@ -0,0 +1,143 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.autoscaler;
+
+enum RayNodeKind {
+  HEAD = 0;
+  WORKER = 1;
+  UNKNOWN = 2;
+}
+
+// A node type that's available for the cluster.
+message InstanceType {
+  bytes type_name = 1;
+  RayNodeKind node_kind = 2;
+  map<string, double> resources = 3;
+}
+
+message GetAvailableInstanceTypesRequest {}
+
+message GetAvailableInstanceTypesResponse {
+  repeated InstanceType instance_types = 1;
+  int64 available_instances = 2;
+}
+
+// Represents the state of a launched instance.
+// An instance is considered launched as long as
+// it has a unique instance_id associated with it.
+//
+// Note a launched instance may be DEAD. In this case,
+// the state will be garbage collected after some timeout
+// period (by default 30 minutes).
+message Instance {
+  enum InstanceStatus {
+    // The unspecified state - most likey it is queued.
+    INSTANCE_STATUS_UNSPECIFIED = 0;
+    // Instance is starting. The first state update received from the
+    // instance.
+    STARTING = 1;
+    // The instance is running - one of two states of a healthy instance.
+    RUNNING = 2;
+    // The instance is idle - one of two states of a healthy instance.
+    IDLE = 3;
+    // The instance is stopping - usually follows from the RUNNING, IDLE,
+    // PREEMPT_REQUEST or DRAIN_REQUEST state.
+    STOPPING = 4;
+    // The instance is stopped - follows from the STOPPING state.
+    STOPPED = 5;
+    // The instance is in a bad state - but it is still able to send updates.
+    FAILING = 6;
+    // The subscribe service moves instances to this state if they
+    // have been idle for too long. This allows the cluster manager to
+    // make a final decision on whether or not to commence a drain
+    // sequence for this instance.
+    DRAIN_CONFIRMATION_PENDING = 7;
+    // The instance should be drained, Ray should start draining process
+    // but could reject if failed to drain.
+    DRAIN_REQUEST = 8;
+    // The instance will be reempted by the instance manager, regardless
+    // of whether it is drainable or not.
+    PREEMPT_REQUEST = 9;
+    // An optional state that can be used to indicate that the instance
+    // is allocated from cloud provider, but ray hasn't been installed yet.
+    INSTANCE_ALLOCATED = 10;
+    // An optional state that can be used to indicate that the instance
+    // is currently installing Ray.
+    INSTALLING_RAY = 11;
+    // An optional state that can be used to indicate that the instance
+    // failed to allocate from cloud provider.
+    ALLOCATION_FAILED = 12;
+    // Node is deleted.
+    GARAGE_COLLECTED = 13;
+  }
+  // an unique id for the instance that's generated by the
+  // instance manager. This may be optional if
+  // the instance hasn't be started yet.
+  string instance_id = 11;
```

Comment:
Do you think it makes sense to make bytes type since that's our standard for all ids in Ray core?